packages/query/src/bitcoin/clients/best-in-slot.ts

```diff
@@ -222,7 +223,44 @@ export function BestinSlotApi() {
 
     const resp = await axios.get<BestinSlotInscriptionByAddressResponse>(
       `${baseUrl}/wallet/inscriptions?${queryParams}`,
-      { ...defaultOptions, signal }
+      { signal }
+    );
+
+    try {
+      return inscriptionsByAddressSchema.parse(resp.data);
+    } catch (e) {
+      // TODO: should be analytics
+      // eslint-disable-next-line no-console
+      if (e instanceof ZodError) console.log('schema_fail', e);
+      throw e;
+    }
+  }
+
+  async function getInscriptionsByAddresses({
+    addresses,
+    network = WalletDefaultNetworkConfigurationIds.mainnet,
+    sortBy = 'inscr_num',
+    order = 'asc',
+    offset = 0,
+    // 2000 is the maximum count
+    count = 100,
+    exclude_brc20 = false,
+    signal,
+  }: BestInSlotInscriptionByAddressesArgs) {
+    const baseUrl = network === 'mainnet' ? url : testnetUrl;
```

Comment:
whenNetwork?

Reply:
there is type error bc of we pass to whenNetwork `NetworkModes`, but here network type is `BitcoinNetworkModes`

Reply:
`bitcoinNetworkModeToCoreNetworkMode` for cases like this